queries/robot/highlights.scm

```diff
@@ -1,21 +1,53 @@
-(argument (dictionary_variable) @string.special)
-(argument (list_variable) @string.special)
-(argument (scalar_variable) @string.special)
-(argument (text_chunk) @string)
+(comment) @comment
 
-(keyword_invocation (keyword) @function)
+(section_header) @keyword
+(extra_text) @comment
 
-(test_case_definition (name) @property)
+(setting_statement) @keyword
+
+(variable_definition (variable_name) @variable)
 
-(keyword_definition (body (keyword_setting) @keyword))
 (keyword_definition (name) @function)
+(keyword_definition (body (keyword_setting) @keyword))
 
-(variable_definition (variable_name) @variable)
+(test_case_definition (name) @property)
 
-(setting_statement) @keyword
+(keyword_invocation (keyword) @function.call)
+(ellipses) @punctuation.delimiter
 
-(extra_text) @comment
-(section_header) @keyword
+(argument (text_chunk) @string)
+(argument (scalar_variable) @variable)
+(argument (list_variable) @variable)
+(argument (dictionary_variable) @variable)
+(argument (inline_python_expression) @string.special)
 
 (ellipses) @punctuation.delimiter
-(comment) @comment
+
+; Control structures
+(for_statement
+  "FOR" @repeat
+  "END" @repeat)
+(for_statement (in "IN" @repeat))
+(for_statement (in_range "IN RANGE" @repeat))
+(for_statement (in_enumerate "IN ENUMERATE" @repeat))
+(for_statement (in_zip "IN ZIP" @repeat))
```

Comment:
Same here, does "FOR" appear in other places? If not, you can just make a group for all of these and mark them as repeat

```scm
[
  "FOR"
  "END"
  "IN"
  ; etc
] @repeat

Reply:
I don't think I can do that, unfortunately Robot has "unquoted strings" and very lose naming rules for keywords (which are basically functions). This means that "END" can either be a built-in keyword, a user-defined keyword or a chunk of a string. Furthermore, the "END" in a loop is a different capture from the "END" of a conditional, which is a different capture from the "END" in a TRY/CATCH statement.

Reply:
Ah, I see. How about leaving the specific "END" capture as is and grouping the others like FOR, IN, etc?

Reply:
> Also, bare identifiers named "END" or "FOR" aren't captured here, it is literal string symbols that are used in the grammar, or error recovery nodes

I'm not sure I followed that. Are you saying that if I have a node (for example `text_chunk`) that contains the text `FOR`, then `"FOR" @repeat` won't match it? :thinking:

Reply:
yes, if it's a terminal symbol that relies on regex

Reply:
To clarify, you're saying `"FOO"` in a query will only match a string that is literally defined as `"FOO"` in the grammar?

In either case, I tested it and it seems to only match the keywords, so all good :+1:

Reply:
yep, exactly

Reply:
@amaanq I updated it now, though to be honest, I liked it better before. It clearly and explicitly followed the syntax of the language. Now it's more jumbled and it feels much easier to make a mistake. :thinking:

Reply:
Tbh, this is subjective for sure, I think in this case you can keep them as you had it before, my *main* want was to have groups of similar nodes (like keyword) grouped together. But since there's going to be a nested capture for the different END types, you can split up the literals like FOR, WHILE, and keep them with their respective nodes too, lua does this.